Rust/Rust.sublime-syntax

```diff
@@ -633,24 +612,67 @@ contexts:
           pop: true
         - match: ','
           scope: punctuation.separator.rust
-        - include: type-any-identifier
+        - include: types-any
     - match: '\+'
       scope: keyword.operator.rust
+    - match: '\b_\b'
+      scope: keyword.operator.rust
+    - match: '!(?!\w)'
+      scope: storage.type.rust
     - match: \bextern\b
       scope: keyword.other.rust
       push:
         - include: strings
         - match: '(?=\S)'
           pop: true
     - include: hrtb
-    - include: type
     - include: type-slice-or-array
-    - match: '\b_\b'
-      scope: keyword.operator.rust
-    - match: '!'
-      scope: keyword.operator.rust
+    - include: type-any-identifiers
+
+  type-any-identifier-path:
+    - meta_scope: meta.path.rust
+    - match: '::'
+      scope: punctuation.accessor.rust
+      pop: true
+    - include: type-identifiers
+
+  type-any-identifiers:
+    - match: '[!?]'
+      scope: storage.modifier.rust
```

Comment:
Can those symbols get a unique subscope, so that they can be distinguished from actual modifier keywords (here `const`, `pub`, `mut`, `ref`, ...), which already use `storage.modifier.rust`? I would want to highlight the `!` and `?` different in my color scheme, just for the reason that they are symbols.

I don't use Rust and don't know about the meaning here, but for example the TypeScript syntax uses `storage.modifier.optional` and `storage.modifier.definite` when those symbols are used as a modifier suffixes. Or PHP uses `storage.type.nullable` for `?`. Perhaps one of these subscopes can be reused, or you can find good alternatives which fit here.